docs/source/api_reference/featurizers.rst

```diff
@@ -195,66 +263,148 @@ lattice and 3D coordinates that specify a periodic crystal structure.
+a Byte-Piece Encoder, which can be found `here.`_
+
+.. _`(transformers.PreTrainedTokenizer)`: https://huggingface.co/transformers/main_classes/tokenizer.html#transformers.PreTrainedTokenizer
+.. _`HuggingFace tokenizers docs`: https://huggingface.co/transformers/main_classes/tokenizer.html
+.. _`here.`: https://github.com/deepchem/deepchem/blob/master/examples/tutorials/22_Transfer_Learning_With_HuggingFace_tox21.ipynb
+
+SmilesTokenizer
+^^^^^^^^^^^^^^^
 
-ElementPropertyFingerprint
-^^^^^^^^^^^^^^^^^^^^^^^^^^
+The :code:`dc.feat.SmilesTokenizer` module inherits from the BertTokenizer class in transformers.
+It runs a WordPiece tokenization algorithm over SMILES strings using the tokenisation SMILES regex developed by Schwaller et. al.
 
-.. autoclass:: deepchem.feat.ElementPropertyFingerprint
+The SmilesTokenizer employs an atom-wise tokenization strategy using the following Regex expression: ::
+
+    SMI_REGEX_PATTERN = "(\[[^\]]+]|Br?|Cl?|N|O|S|P|F|I|b|c|n|o|s|p|\(|\)|\.|=|#||\+|\\\\\/|:||@|\?|>|\*|\$|\%[0–9]{2}|[0–9])"
+
+
+To use, please install the transformers package using the following pip command: ::
+
+    pip install transformers
+
+References:
+
+-  `RXN Mapper: Unsupervised Attention-Guided Atom-Mapping`_
+-  `Molecular Transformer: Unsupervised Attention-Guided Atom-Mapping`_
+
+.. autoclass:: deepchem.feat.SmilesTokenizer
   :members:
 
-ElemNetFeaturizer
-^^^^^^^^^^^^^^^^^^^^^
+BasicSmilesTokenizer
+^^^^^^^^^^^^^^^^^^^^
 
-.. autoclass:: deepchem.feat.ElemNetFeaturizer
+The :code:`dc.feat.BasicSmilesTokenizer` module uses a regex tokenization pattern to tokenise SMILES strings.
+The regex is developed by Schwaller et. al. The tokenizer is to be used on SMILES in cases
+where the user wishes to not rely on the transformers API.
+
+References:
+
+-  `Molecular Transformer: Unsupervised Attention-Guided Atom-Mapping`_
+
+.. autoclass:: deepchem.feat.BasicSmilesTokenizer
   :members:
 
+.. _`RXN Mapper: Unsupervised Attention-Guided Atom-Mapping`: https://chemrxiv.org/articles/Unsupervised_Attention-Guided_Atom-Mapping/12298559
+.. _`Molecular Transformer: Unsupervised Attention-Guided Atom-Mapping`: https://pubs.acs.org/doi/10.1021/acscentsci.9b00576
+
+Other Featurizers
+-----------------
+
 BindingPocketFeaturizer
------------------------
+^^^^^^^^^^^^^^^^^^^^^^^
 
 .. autoclass:: deepchem.feat.BindingPocketFeaturizer
   :members:
+  :inherited-members:
 
 UserDefinedFeaturizer
----------------------
+^^^^^^^^^^^^^^^^^^^^^
 
 .. autoclass:: deepchem.feat.UserDefinedFeaturizer
   :members:
+  :inherited-members:
 
-BPSymmetryFunctionInput
------------------------
+Base Featurizers (for develop)
+------------------------------
 
-.. autoclass:: deepchem.feat.BPSymmetryFunctionInput
+Featurizer
+^^^^^^^^^^
+
+The :code:`dc.feat.Featurizer` class is the abstract parent class for all featurizers.
+
+.. autoclass:: deepchem.feat.Featurizer
   :members:
 
-RawFeaturizer
--------------
+MolecularFeaturizer
+^^^^^^^^^^^^^^^^^^^
```

Comment:
For each of the abstract featurizers, can we add in a sentence explaining when to use this featurizer? For example:

"""
If you're creating a new featurizer that featurizes molecules, you will want to inherit from the abstract `MolecularFeaturizer` base class.
"""

and so on

Reply:
I agree! I think the best practical API design is to have MolecularFeaturizer be the abstract base class rather then creating a new separate module for tokenizers. The only time I can see that being useful is if more varying tokenization algorithms for protein sequences, SMILES, SELFIES, etc, come out and until then having them under 'Featurizers' makes the most sense.

Reply:
> For each of the abstract featurizers, can we add in a sentence explaining when to use this featurizer?

I added!